liferay-echo/liferay-portal
# LPS 190225 Discard deleted fragment entry links when load/reload a page

Motivation
=======
As part of [story](https://liferay.atlassian.net/browse/LPS-190225) related with undo/redo we mark fragments as deleted, to make it easier the redo. Since, when we reload the page the undo/redo is lost, we are loading deleted FragmentEntryLinks and it is not necessary.


Proposed Solution
========
The idea of this task is doesn’t load deleted fragments in `_getFragmentEntryLinks().`

Steps to Verify
========
- Add a breakpoint in ContentPageEditorDisplayContext.java line 722 and connect the debugger.

```
).put(
	"fragmentEntryLinks", _getFragmentEntryLinks().  <--- Here
).put(

```

- Go to the portal and create a new content page.
- Add 3 buttons and remove 2 of them using the **UNDO** arrows on the top right part of the page.
- Without saving or publishing the changes, reload the page using the **reload button in your browser**.
- Then you debugger should stop at the break point and when evaluating result of the` _getFragmentEntryLinks()` method, only one _FragmentEntryLink_ should be showed. (if you repeat this code in master, without this changes, you'll be able to see multiple _FragmentEntryLinks_ in te response as the deleted ones are included as well).

** please let me know if you need any help to verify it
